template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplateModelContextBuilder.java

```diff
@@ -162,4 +162,11 @@ public TemplateModelContextBuilder withFixInputs(Map<String, Object> fixInputs)
         blueprintTemplateModelContext.put(HandleBarModelKey.STACK_VERSION.modelKey(), "{{stack_version}}");
         return blueprintTemplateModelContext;
     }
+
+    private Map<String, Object> createAdjustedFileSystemConfig() {
+        Map<String, Object> result = new HashMap<>();
+        result.putAll(fileSystemConfig);
+        result.put("cloudStorageEnabled", !fileSystemConfig.isEmpty());
```

Comment:
This has the potential to byte us for the on-prem use case (Central HDFS, workload clusters do not have an HDFS instance, and are instead configured to point to the central HDFS) or when Ozone comes into play.

That said, I'd imagine the on-prem use case fits very nicely into this model. The name could cause some confusion. Looks good for now.